### PR TITLE
ci: add prevent deletion

### PR DESCRIPTION
**Pull request by Oreoxmt**
### What is changed, added or deleted? (Required)

- Background: it is **NOT** allowed to delete images on any branches.
- Add a new workflow to check whether an image was deleted in PRs.
- If you delete/rename images in a PR: 
   image

  [click Details](https://github.com/pingcap/docs/pull/10156/checks?check_run_id=8007876325) to get the list of deleted images.
  image

- Otherwise: [details](https://github.com/pingcap/docs/pull/10156/checks?check_run_id=8007857776)
  image

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [x] v5.4 (TiDB 5.4 versions)
- [x] v5.3 (TiDB 5.3 versions)
- [x] v5.2 (TiDB 5.2 versions)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: .github/workflows/prevent-deletion.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      checks: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Check changes
+        run: |
+          if git diff --name-only --diff-filter 'D' '${{ github.event.pull_request.base.sha }}' '${{ github.event.pull_request.head.sha }}' | grep -E '^media/.*\.(jpg|png|jpeg)$' >/tmp/changed_files; then
```
Comment (ran-huang) [inline .github/workflows/prevent-deletion.yaml:18]:
```suggestion
          if git diff --name-only --diff-filter 'D' '${{ github.event.pull_request.base.sha }}' '${{ github.event.pull_request.head.sha }}' | grep -E '^media/.*\.(jpg|png|jpeg|gif)$' >/tmp/changed_files; then
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM.

**Review by TomShawn — APPROVED** [review]

**Pull request merged**